File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        anam = {}
        if len(s) != len(t):
            return False
        contagem = {}
        for letra in s:
            
            contagem[letra] = contagem.get(letra, 0) + 1
        for letra in t:
            if contagem.get(letra, 0) == 0:
                return False
            
            contagem[letra] -= 1

        return True
            

File: 0242-valid-anagram/0242-valid-anagram.java
import java.util.HashMap;

class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }

        HashMap<Character, Integer> contagem = new HashMap<>();
        for (char c : s.toCharArray()) {
            contagem.put(c, contagem.getOrDefault(c, 0) + 1);
        }
        for (char c : t.toCharArray()) {
            int atual = contagem.getOrDefault(c, 0);
            
            if (atual == 0) {
                return false; 
            }
            
            contagem.put(c, atual - 1);
        }

        return true;
    }
}